File: leetcode/chain/83.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:

    def deleteDuplicates(self, head: ListNode) -> ListNode:
        '''删除排序链表中的重复值'''
        ret = head
        while head is not None:
            if head.next is not None and head.val == head.next.val:
                head.next = head.next.next
            else:
                head = head.next
        return ret

File: leetcode/chain/83.cpp
#include <iostream>

#include "list_node.h"

class Solution {
 public:
  ListNode* delete_duplicates(ListNode* node) {
    ListNode *ret = node;
    while (node != nullptr) { // 遍历
      while (node->next != nullptr && node->val == node->next->val) { // 去重
        node->next = node->next->next;
      }
      node = node->next;
    }
    return ret;
  }
};

int main() {
  ListNode* node = new ListNode(3);
  node->append(new ListNode(4));
  node->append(new ListNode(4));
  node->append(new ListNode(5));
  node->print();

  Solution solution = Solution();
  ListNode* ret = solution.delete_duplicates(node);
  ret->print();
}